src/handlers/handler_token.py

```diff
@@ -0,0 +1,142 @@
+
+logger = logging.getLogger(__name__)
+log_level = os.environ.get("LOG_LEVEL", "INFO")
+logger.setLevel(log_level)
+
+
+class HandlerToken:
+    """
+    HandlerToken manages token provider URL and public keys for JWT verification.
+    """
+
+    def __init__(self, config):
+        self.provider_url: str = config.get(TOKEN_PROVIDER_URL, "")
+        self.public_keys_url: str = config.get(TOKEN_PUBLIC_KEYS_URL) or config.get(TOKEN_PUBLIC_KEY_URL)
+        self.public_keys: list[RSAPublicKey] = []
+
+    def load_public_keys(self) -> "HandlerToken":
+        """
+        Load token public keys from the configured URL.
+        Returns:
+            HandlerToken: The current instance with loaded public keys.
+        Raises:
+            RuntimeError: If fetching or deserializing the public keys fails.
+        """
+        logger.debug("Loading token public keys from %s", self.public_keys_url)
+
+        try:
+            response_json = requests.get(self.public_keys_url, verify=False, timeout=5).json()
+            raw_keys: list[str] = []
+
+            if isinstance(response_json, dict):
+                if "keys" in response_json and isinstance(response_json["keys"], list):
+                    for item in response_json["keys"]:
+                        if "key" in item:
+                            raw_keys.append(item["key"].strip())
+                elif "key" in response_json:
+                    raw_keys.append(response_json["key"].strip())
+
+            if not raw_keys:
+                raise KeyError(f"No public keys found in {self.public_keys_url} endpoint response")
+
+            self.public_keys = [
+                cast(RSAPublicKey, serialization.load_der_public_key(base64.b64decode(raw_key))) for raw_key in raw_keys
+            ]
+            logger.debug("Loaded %d token public keys", len(self.public_keys))
+
+            return self
+        except (requests.RequestException, ValueError, KeyError, UnsupportedAlgorithm) as exc:
+            logger.exception("Failed to fetch or deserialize token public key from %s", self.public_keys_url)
+            raise RuntimeError("Token public key initialization failed") from exc
+
+    def decode_jwt(self, token_encoded: str) -> Dict[str, Any]:
+        """
+        Decode and verify a JWT using the loaded public keys.
+        Args:
+            token_encoded (str): The encoded JWT token.
+        Returns:
+            Dict[str, Any]: The decoded JWT payload.
+        Raises:
+            jwt.PyJWTError: If verification fails for all public keys.
+        """
+        logger.debug("Decoding JWT")
+        for public_key in self.public_keys:
+            try:
+                return jwt.decode(token_encoded, public_key, algorithms=["RS256"])
+            except jwt.PyJWTError:
+                continue
+        raise jwt.PyJWTError("Verification failed for all public keys")
+
+    def get_token(self) -> Dict[str, Any]:
```

Comment:
`get_token_provider_info` maybe, or something similar? `get_token` suggests actual token of some kind is returned, at least to me.